mlflow/utils/autologging_utils/__init__.py

```diff
@@ -429,3 +429,86 @@ def autologging_is_disabled(integration_name):
         return get_autologging_config(integration_name, "disable_for_unsupported_versions", False)
 
     return False
+
+
+def _get_training_session():
+    """
+    Returns a session manager for nested autologging runs.
+    """
+    # NOTE: The current implementation doesn't guarantee thread-safety, but that's okay for now
+    # because:
+    # 1. We don't currently have any use cases for allow_children=True.
+    # 2. The list append & pop operations are thread-safe, so we will always clear the session stack
+    #    once all _SklearnTrainingSessions exit.
+    class _TrainingSession(object):
```

Comment:
Created this function to avoid using the same session manager across different flavors.